src/browser/components/ClipboardCopier.tsx

```diff
@@ -0,0 +1,76 @@
+import styled from 'styled-components'
+import React, { useState } from 'react'
+import { CopyIcon } from './icons/Icons'
+
+type ClipboardCopierProps = {
+  textToCopy: string
+  iconSize?: number
+}
+function ClipboardCopier({
+  textToCopy: text,
+  iconSize = 20
+}: ClipboardCopierProps): JSX.Element {
+  const [messageToShow, setMessageToShow] = useState<string | null>(null)
+  function showPopup(text: string) {
+    setMessageToShow(text)
+    setTimeout(() => setMessageToShow(null), 1500)
+  }
+
+  return (
+    <CopyIconContainer
+      onClick={() =>
+        copyToClipboard(text)
+          .then(() => showPopup('✔️ Copied to clipboard'))
+          .catch(() => showPopup('Copying text failed'))
+      }
+    >
+      <CopyIcon title="Copy to clipboard" width={iconSize} />
+      {messageToShow && <InfoPopup text={messageToShow} />}
+    </CopyIconContainer>
+  )
+}
+
+const CopyIconContainer = styled.span`
+  cursor: pointer;
+  position: relative;
+`
+
+type InfoPopupProps = { text: string }
+function InfoPopup({ text }: InfoPopupProps) {
+  return <PopupTextContainer> {text} </PopupTextContainer>
+}
+const PopupTextContainer = styled.span`
+  position: absolute;
+  white-space: nowrap;
+  right: 30px;
+  bottom: 0;
+  border-radius: 2px;
+  background-color: ${props => props.theme.frameSidebarBackground};
+  box-shadow: ${props => props.theme.standardShadow};
+  color: ${props => props.theme.primaryText}
+  font-family: ${props => props.theme.drawerHeaderFontFamily};
+  padding: 0 5px;
+`
+
+export function copyToClipboard(text: string): Promise<void> {
```

Comment:
nice util function, as you said it might be good to eventually move this to a separate utils file or something but for now it's fine to keep it here.